### DISCUSSION TITLE
How do I get pagefind to work with the nextjs app router?

### DISCUSSION THREAD

**Discussion by ytsmash** [Pagefind/pagefind#612] [Q&A]
My website is SSG with nextjs app router.
I checked the documentation and could only find the followind for pages router.

https://www.petemillspaugh.com/nextjs-search-with-pagefind

Please tell me how to introduce Pagefind into the app router 

## version

nextjs: 14.2.3
nodejs: 18.17.1

**Comment by bglw** [comment]
Hi @ytsmash 👋 

I don't have personal experience with Next.js and Pagefind, so I'll only be able to point you toward others that have written up their solutions for this.

- The blog you linked is often a good source for people
- https://lib.rs/crates/next-pagefind is something I have seen but not used
- Chucking a comment/question in [this discussion](https://github.com/CloudCannon/pagefind/discussions/481) might yield some answers

Sorry to not have anything specific for the app router, not something I have used personally. Hopefully someone out there who has used Pagefind and Next's app router can chime in and help out. 

I'll convert this over to a discussion so it can stick around longer.
